rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,896 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="hsearch" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset provides analysis for migration from Hibernate Search 4.x to Hibernate Search 5.x.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,2.4.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,2.4.0.Final" />
+        </dependencies>
+        <sourceTechnology id="hibernate-search" versionRange="[4,)" />
+        <targetTechnology id="hibernate-search5" versionRange="[5,)" />
+        <targetTechnology id="eap7" versionRange="[7,)" />
+        <tag>reviewed-2016-04-12</tag>
+    </metadata>
+    <rules>
+        <rule id="hsearch-00000">
+            <when>
+                <javaclass references="org.hibernate.search.{*}" />
+            </when>
+            <perform>
+                <classification title="Hibernate Search usage" effort="0" severity="mandatory">
+                    <description>Hibernate Search API used.</description>
+                    <tag>hibernate-search</tag>
+                </classification>
+            </perform>
+        </rule>
+        <rule id="hsearch-00001">
+            <when>
+                <javaclass references="org.hibernate.search.engine.impl.SearchMappingBuilder" />
+            </when>
+            <perform>
+                <hint title="Removed org.hibernate.search.engine.impl.SearchMappingBuilder class in Hibernate Search 5" effort="1" severity="mandatory">
+                    <message>The class `org.hibernate.search.engine.impl.SearchMappingBuilder` were replaced by `org.hibernate.search.engine.spi.SearchMappingBuilder` class.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_changes"
+                        title="Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00002">
+            <when>
+                <javaclass references="org.hibernate.search.Environment" />
```

Comment:
This class is just a bag of `public static final String`s. Those are compile-time constants, so the references will be replaced by the string values during compilation. Which means that this rule will never match when analyzing compiled applications, it only makes sense in `--sourceMode`. Not sure what to do about that, if something.